### PR TITLE
VACMS-6874: prepopulate entity reference fields on nonclinical service form when context is known

### LINKED ISSUE #6874: Prepopulate entity reference fields on nonclinical service form, when context is known
## User Story or Problem Statement

As an editor of a Top task page that clicks a link to add a non-clinical service, i want the form for non-clinical service to carry some of the context (Service and Section) from the page i was just on. 

[This coudl use some screenshots and URLs in the future]

## Acceptance Criteria
- [ ] Editors that click on the /node/add/vha_facility_nonclinical_service links from within the top taks forms, get Section and Service fields prepopulated on that form, based on parameters on the page from which they are coming.

## Implementation steps

Add prepopulate module (?) so that we can make the “Add a service” link in the header of these views go ahead and prepopulate 2 of the fields on the non-clinical health service form.

Update links in headers on each of the three displays on non_clinical_services to properly prepopulate the fields on the /node/add forms.

## Design principles

Veteran-centered
- [ ] `Single source of truth`: Increase reliability and consistency of content on VA.gov by providing a single source of truth.
- [ ] `Accessible, plain language`: Provide guardrails and guidelines to ensure content quality.
- [ ] `Purposely structured content`: Ensure Content API can deliver content whose meaning matches its structure.
- [ ] `Content lifecycle governance`: Produce tools, processes and policies to maintain content quality throughout its lifecycle.

Editor-centered
- [ ] `Purpose-driven`: Create an opportunity to involve the editor community in VA’s mission and content strategy goals.
- [x] `Efficient`: Remove distractions and create clear, straightforward paths to get the job done.
- [ ] `Approachable`: Offer friendly guidance over authoritative instruction.
- [ ] `Consistent`: Reduce user’s mental load by allowing them to fall back on pattern recognition to complete tasks.
- [ ] `Empowering`: Provide clear information to help editors make decisions about their work.

### CMS Team

Please leave only the team that will do this work selected. If you're not sure, it's fine to leave both selected.

- [ ] `Platform CMS Team`
- [x] `Sitewide CMS Team`

### PR DESCRIPTION
**Pull request by JayDarnell**
## Description

Closes #6874 

## Testing done

Visual Testing

## Screenshots
In the following examples all content is part of the VA Alaska health care section:

From the edit view of a given VAMC System Billing and Insurance node:
Screen Shot 2021-12-15 at 9 37 33 AM

After clicking the **Add a billing and insurance office** link in the image above:
Screen Shot 2021-12-15 at 9 38 28 AM

From the edit view of a given VAMC System Medical Records Office node:
Screen Shot 2021-12-15 at 9 38 37 AM

After clicking the **Add a medical records office** link in the image above:
Screen Shot 2021-12-15 at 9 38 56 AM

From the edit view of a given VAMC System Register for Care node:
Screen Shot 2021-12-15 at 9 39 08 AM

After clicking the **Add an admissions office** link in the image above:
Screen Shot 2021-12-15 at 9 39 27 AM

## QA steps

As an admin user
Billing and Insurance
- [x] Create a VAMC System Billing and Insurance node https://pr7292-e6gssk9rqwby2ud2suwiqdqqfcfppzow.ci.cms.va.gov/node/add/vamc_system_billing_insurance - save the new node
- [x] Edit the node above, scroll to the bottom of the edit page - you should see a link to **Add a billing and insurance office** Click this link and a new tab should open up. Switch to this new tab.
- [x] In this new tab the service field should say Billing and Insurance and the section should match the section for the VAMC System Billing and Insurance node above

Medical Records Office
- [x] Create a VAMC System Medical Records Office node - save the new node
- [x] Edit the node above, scroll to the bottom of the edit page - you should see a link to **Add a medical records office** Click this link and a new tab should open up. Switch to this new tab.
- [x] In this new tab the service field should say Medical Records and the section should match the section for the VAMC System Medical Records Office node above

Register for Care
- [x] Create a VAMC System Register for Care node - save the new node
- [x] Edit the node above, scroll to the bottom of the edit page - you should see a link to **Add an admissions office** Click this link and a new tab should open up. Switch to this new tab.
- [x] A new tab should open up. In this new tab the service field should say Register for Care and the section should match the section for the VAMC System Register for Care node above

### Definition of Done

- [ ] Documentation has been updated, if applicable.
- [ ] Tests have been added if necessary.
- [ ] Automated tests have passed.
- [ ] Code Quality Tests have passed.
- [ ] Acceptance Criteria in related issue are met.
- [ ] Manual Code Review Approved.

### Select Team for PR review

- [ ] `Platform CMS team`
- [x] `Sitewide CMS Team`

### CODE REVIEW

File: config/sync/views.view.non_clinical_services.yml [hunk 1]
```diff
@@ -897,11 +960,12 @@ display:
...
         - user
         - 'user.node_grants:view'
-      tags: {  }
+      tags:
+        - 'config:field.storage.node.field_administration'
   billing_and_insurance:
     display_plugin: block
     id: billing_and_insurance
-    display_title: 'Billing and insurance offices'
+    display_title: 'Billing and Insurance Offices'
```
Comment (swirtSJW) [inline config/sync/views.view.non_clinical_services.yml:968]: Sentence case?  (sorry if this has already been cleared)  I didn't see anything referenced in the issue related to switching away from sentence case.

File: config/sync/views.view.non_clinical_services.yml [hunk 2]
```diff
@@ -990,7 +1054,7 @@ display:
         operator: AND
         groups:
           1: AND
-      title: 'Billing and insurance offices'
+      title: 'Billing and Insurance Offices'
```
Comment (swirtSJW) [inline config/sync/views.view.non_clinical_services.yml:1057]: sentence case

File: config/sync/views.view.non_clinical_services.yml [hunk 3]
```diff
@@ -1350,11 +1476,12 @@ display:
...
         - user
         - 'user.node_grants:view'
-      tags: {  }
+      tags:
+        - 'config:field.storage.node.field_administration'
   medical_records_offices:
     display_plugin: block
     id: medical_records_offices
-    display_title: 'Medical records offices'
+    display_title: 'Medical Records Offices'
```
Comment (swirtSJW) [inline config/sync/views.view.non_clinical_services.yml:1484]: sentence case

**Review by swirtSJW — CHANGES_REQUESTED** [review]
Conflict has this not able to build a pr environment.

**Review by swirtSJW — CHANGES_REQUESTED** [review]
Can't perform second step of 
`Edit the node above, scroll to the bottom of the edit page - you should see a link to Add a billing and insurance office Click this link and a new tab should open up. Switch to this new tab.` 
The link to add is not there 
image

**Review by swirtSJW — CHANGES_REQUESTED** [review]
Also looks like you need to remove the sort link from a column in the table that has no heading.    We actually shouldn't have a column that has no heading.   Thats a 508 thing that is separate form the link with no text.  
image

The sortby on the edit column needs to be removed.  
image


This is actually a UI accessibility failure that node link report just happened to catch.  We are venturing into new ground where UI is existing on node view.... so node link report is going to catch some stuff that is UI related and not content.

**Review by swirtSJW — COMMENTED** [review]
Two issues off the bat 
image

image

I can address these while I fix  the merge conflict.

**Review by swirtSJW — APPROVED** [review]
These test out as working fine.  Nice solution Jay. 

There are two issues I see but I don't think either of them are blockers unless we want them to be.

1) When you click the "add office link" before you save the initial top task page, the pre-filling does not work.  
image
I think there is probably no clean way around this without using a bunch of hacky js.  I am not sure it is worth it.   If it is worth it, we could either
 a) make JS update the link on blur of the office and field_administration field
 b) Not display that "add office" link until the node has been saved  (on node/edit as opposed to node add.

2) The other issue that troubles me is the flip flopping of language on what we are creating.
image
Is it an office, or a service?  Trying to refer to it as both... is going to be a nightmare for training and helpdesk, not to mention understanding.
image
Every place else it is a service.

I'll marked this as approved, but leave it up to @kevwalsh 
as to whether we want to solve this now  or later.  My guess is probably later.

**Comment by kevwalsh** [comment]
The UX writing work is being taken care of elsewhere. We will be moving away from tables, too. Spec coming in #6877.

**Comment by swirtSJW** [comment]
Also a little concerned about 
image

**Comment by swirtSJW** [comment]
image

**Comment by JayDarnell** [comment]
Yeah a lot of the view language was already present and I left it untouched given the fact it was all going to be reworked in the next steps.

**Pull request merged**